### PR TITLE
CATROID-1182 Back button in topbar not working

### PR DESCRIPTION
**Pull request by Tamdav**
https://jira.catrob.at/browse/CATROID-1182

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/BackPressedTest.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        project = Project(
+            ApplicationProvider.getApplicationContext(),
+            Utils.sanitizeFileName(defaultProjectName)
+        )
+        XstreamSerializer.getInstance().saveProject(project)
+        val sprite = Sprite("testSprite")
+        project!!.defaultScene.addSprite(sprite)
```
Comment (khushal1707) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/BackPressedTest.kt:68]:
```suggestion
        project?.defaultScene?.addSprite(sprite)
```

File: catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/BackPressedTest.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        baseActivityTestRule.launchActivity(null)
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun tearDown() {
+        TestUtils.deleteProjects(project!!.name)
```
Comment (khushal1707) [inline catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/BackPressedTest.kt:75]:
```suggestion
        TestUtils.deleteProjects(project?.name)
```

**Comment by khushal1707** [comment]
Just some minor changes, otherwise LGTM!

**Review by khushal1707 — APPROVED** [review]
LGTM!

**Comment by blindguardian50** [comment]
This issue was already solved in https://github.com/Catrobat/Catroid/pull/4266

**Comment by wslany** [comment]
Fixed and merged in #4266.

**Pull request closed without merging**